src/swsssdk/scripts/sonic-db-cli

```diff
@@ -36,3 +57,29 @@ else:
             print "\n".join(resp)
         else:
             print resp
+
+def main():
+    parser = argparse.ArgumentParser(description='SONiC DB CLI',
+                         formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('db_or_op', nargs='?', type=str, help='Database name Or Unary operation')
+    parser.add_argument('cmd', nargs='*', type=str, help='Command to execute in database')
+
+    args = parser.parse_args()
+
+    if args.db_or_op:
+        if args.cmd:
+            execute_cmd(args.db_or_op, args.cmd)
+        elif args.db_or_op == 'PING':
+            ping_all_instances()
```

Comment:
>ping_all_instances [](start = 12, length = 18)

What is the expected behaviors?
1. redis-server is running and pingable
2. redis-server is running but not pingable
3. redis-server is not running.

Reply:
For example we have four instances. 

1  all redis-server is running and pingable, return a simple 'PONG' which mimic redis-cli, the application who calling sonic-db-cli will check the return value
 
2 redis-server is running but not pingable
3 redis-server is not running.
For thesw two failed cases, at least one redis-server is failed to ping. We return the results list with error msg. The msg is not 'PONG' only and the application knows ping all servers failed.

e.g.
```
## FAILED cases
admin@ASW-7005:~$ sonic-db-cli PING
['PONG', 'PONG', ConnectionError(u'Error 111 connecting to 127.0.0.1:36379. Connection refused.',), 'PONG']

##SUCCESS cases
admin@ASW-7005:~$ sonic-db-cli PING
PONG
```

Reply:
I see original usage
```
until [[ $(redis-cli ping | grep -c PONG) -gt 0 ]]; do
```

In your example case:
```
['PONG', 'PONG', 'Error 111 connecting to 127.0.0.1:36379. Connection refused.', 'PONG']
```
will be treated as passed.

---
In reply to: [405730130](https://github.com/Azure/sonic-py-swsssdk/pull/65#discussion_r405730130) [](ancestors = 405730130)

Reply:
I think we should make the printout msg simple as what it has for SUCCESS case. Only a simple 'FAILED' is enough. We only wonder it success or not.

Reply:
```
admin@ASW-7005:~$ /tmp/sonic-db-cli PING
PONG
admin@ASW-7005:~$ 

admin@ASW-7005:~$ /tmp/sonic-db-cli PING
FAILED
admin@ASW-7005:~$ 
```

Reply:
To mimic redis-cli is a good idea. I just experiment with single instance, and see the failure case has some error messages and return code. Could you please mimic them?

```
$ redis-cli ping
PONG
$ sudo service redis-server stop
Stopping redis-server: redis-server.
$ redis-cli ping
Could not connect to Redis at 127.0.0.1:6379: Connection refused
Could not connect to Redis at 127.0.0.1:6379: Connection refused
$ echo $?
1
```


---
In reply to: [407084792](https://github.com/Azure/sonic-py-swsssdk/pull/65#discussion_r407084792) [](ancestors = 407084792,405730130)

Reply:
Make the output and return code the same as redis-cli

```
admin@ASW-7005:~$ sonic-db-cli PING
PONG
admin@ASW-7005:~$ echo $?
0

admin@ASW-7005:~$ sonic-db-cli PING
Could not connect to Redis at 127.0.0.1:116379: Connection refused
Could not connect to Redis at 127.0.0.1:126379: Connection refused
admin@ASW-7005:~$ echo $?
1
```